### DISCUSSION TITLE
wonder how can add more markdown syntax support in the visual editor?

### DISCUSSION THREAD

**Discussion by tx0c** [remirror/remirror#1514] [Q&A] [answered]
for example in the example editor: https://remirror.vercel.app/?path=/story/editors-markdown--basic

if I type in `**bold**` then it automatically becomes bold text,

![image](https://user-images.githubusercontent.com/49659410/153107861-e620204d-5372-4ae0-b260-b81001e2817f.png)

but if I'm trying some more advanced markdown syntax, like link or image, nothing happens there,

```
![image](https://user-images.githubusercontent.com/49659410/153107861-e620204d-5372-4ae0-b260-b81001e2817f.png)
```

Wonder how can add more markdown syntax support in the visual editor? to also support when people copy and paste a well-formatted markdown from somewhere else,

**Comment by labkey-alan** [comment, marked as answer]
To enable this you'll need to create your own extension that has a pasteRule and an input rule that detect this form of Markdown syntax and convert it to an image. If you take a look at the ItalicExtension you'll see an example of paste rules and input rules.
